plan/aggregation_pruning.go

```diff
@@ -29,7 +29,8 @@ type aggPruner struct {
 // eliminateAggregation will eliminate aggregation grouped by unique key.
 // e.g. select min(b) from t group by a. If a is a unique key, then this sql is equal to `select b from t group by a`.
 // For count(expr), sum(expr), avg(expr), we may need to rewrite the expr. Details is shown below.
-func (ap *aggPruner) eliminateAggregation(p LogicalPlan) error {
+func (ap *aggPruner) eliminateAggregation(p LogicalPlan) (LogicalPlan, error) {
+	newChildren := make([]Plan, 0, len(p.GetChildren()))
 	for _, child := range p.GetChildren() {
```

Comment:
If you return LogicalPlan, you can call eliminateAggregation for every child. And only process `ap` here.